### DISCUSSION TITLE
Disable border for inactive window clients

### DISCUSSION THREAD

**Discussion by sghol** [qtile/qtile#4103] [Q&A]
I don't see any reason to have a border color for inactive client windows. Is there anyway to have a border only for active(focused) client window?

Why we cannot do something like this:
```python
    layout.Column(
          ...
          border_normal = None
          ...
)


```
I try hooks but I didn't get succeeded 
```python
windows = None

@hook.subscribe.group_window_add
def on_new_window(group, window):
    global windows
    windows = group.windows


@hook.subscribe.client_focus
def on_client_focus(window):
    global windows
    for win in windows:
        win.borderwidth = 0  #   it doesn't change border for any window
```

**Comment by m-col** [comment]
>           border_normal = None`

This won't work because this option only accepts a colour. Alpha channels are supported though, so you could pass `"#00000000"` and the border will be invisible (may need a compositor if you're running in X11).

If you do really want the window to fill the space where the borders would be, i.e. with a width of 0, your best bet would be to subclass your layout and add custom logic:
```py
class MyColumns(layout.Columns):
    def configure(self, client, screen_rect):
        pos = 0
        for col in self.columns:
            if client in col:
                break
            pos += col.width
        else:
            client.hide()
            return
        if client.has_focus:
            color = self.border_focus if col.split else self.border_focus_stack
            border = self.border_width
        else:
            color = self.border_normal if col.split else self.border_normal_stack
            border = 0
        margin_size = self.margin
        if len(self.columns) == 1 and (len(col) == 1 or not col.split):
            if not self.border_on_single:
                border = 0
            if self.margin_on_single is not None:
                margin_size = self.margin_on_single
        width = int(0.5 + col.width * screen_rect.width * 0.01 / len(self.columns))
        x = screen_rect.x + int(0.5 + pos * screen_rect.width * 0.01 / len(self.columns))
        if col.split:
            pos = 0
            for c in col:
                if client == c:
                    break
                pos += col.heights[c]
            height = int(0.5 + col.heights[client] * screen_rect.height * 0.01 / len(col))
            y = screen_rect.y + int(0.5 + pos * screen_rect.height * 0.01 / len(col))
            client.place(
                x, y, width - 2 * border, height - 2 * border, border, color, margin=margin_size
            )
            client.unhide()
        elif client == col.cw:
            client.place(
                x,
                screen_rect.y,
                width - 2 * border,
                screen_rect.height - 2 * border,
                border,
                color,
                margin=margin_size,
            )
            client.unhide()
        else:
            client.hide()
```
This looks more significant than it is - all I did was copy the `def configure` from the original layout and move the `border =` line into the `client.has_focus` if block to conditionally set the border width.

**Reply by sghol** [reply]
Thanks for the response.
I don't like the inheritance approach because it's too much just for disabling the border. I change the original source code of qtile and make the border to zero but I didn't like it because moving between window clients looked wired!
But I like the way we specify alpha value to the color of the border.
Unfortunately, I didn't work. I have a picom as a compositor manager, I give color value with alpha value but nothing happened to the border!
``` python
...
border_normal = "#00000000",
..
```
 I guess that borders don't care alpha value.  I test alpha value on qtile's bar and it's worked perfectly.

Picom has a option named `frame-opactiy`, it adds transparency to the border of the window clients, but the problem is it's add value to all window clients rather than active or inactive. So if I find an option in picom which only add opacity to the border of the inactive window clients my problem will solve without touching my qtile configs.
